app/controllers/candidate_interface/other_qualifications/details_controller.rb

```diff
@@ -50,26 +40,15 @@ def create
 
     def other_qualification_params
       params.require(:candidate_interface_other_qualification_form).permit(
-        :id, :subject, :institution_name, :grade, :award_year, :choice
-      ).merge!(id: params[:id], qualification_type: get_qualification_type).transform_values(&:strip)
-    end
-
-    def last_two_qualifications_are_of_same_type(qualifications)
-      second_to_last_qualification = qualifications[-2]
-      last_qualification = qualifications[-1]
-      second_to_last_qualification&.qualification_type == last_qualification.qualification_type
-    end
-
-    def pre_fill_institution_name(qualifications)
-      qualifications[-2].institution_name
-    end
-
-    def pre_fill_award_year(qualifications)
-      qualifications[-2].award_year
+        :id, :subject, :institution_name, :grade, :award_year, :choice, :institution_country
+      ).merge!(id: params[:id],
+               qualification_type: get_qualification.qualification_type,
+               non_uk_qualification_type: get_qualification.non_uk_qualification_type,
+               other_uk_qualification_type: get_qualification.other_uk_qualification_type)
     end
 
-    def get_qualification_type
-      ApplicationQualification.find(params[:id]).qualification_type
+    def get_qualification
+      ApplicationQualification.find(params[:id])
```

Comment:
Every call to this method will hit the database, I think. We could [memoize](https://en.wikipedia.org/wiki/Memoization) it:

```suggestion
      @get_qualification ||= ApplicationQualification.find(params[:id])
```